### PR TITLE
data generation script completed

### PR DESCRIPTION
**Pull request by rogen-code**

### CODE REVIEW

File: generator.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const fs = require('fs')
+const faker = require('faker')
+const argv = require('yargs').argv
```
Comment (JacksonTodd93) [inline generator.js:3]: Just read some of the docs for this library. Sounds cool!

File: helpers/helpers.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports.generateRandomImage = function() {
+  let first = Math.floor(Math.random() * 9)
+  let second = Math.floor(Math.random() * 9)
+  let third = Math.floor(Math.random() * 9)
+  return `${first}${second}${third}`
```
Comment (JacksonTodd93) [inline helpers/helpers.js:5]: Have you thought about appending your file type extension here?  (.jpg, .png, etc).

File: generator.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const helpers = require('./helpers/helpers.js')
+
+const residences = argv.residences || 10000000
+const filename = argv.output || 'residences.csv'
+const stream = fs.createWriteStream(filename)
+
+const createResidence = () => {
+  const residenceName = faker.company.companyName();
+  const price = faker.commerce.price();
+  const url = helpers.generateRandomImage();
```
Comment (JacksonTodd93) [inline generator.js:13]: Two things come to mind for this:
1) Am I correct in interpreting that this means each residence will only ever one photo attached to it?
2) I think you could be slightly more precise in naming this function. You're not generating a random image exactly, you're generating a random number that corresponds to the URLs of images in your S3 database.

File: schema.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*  Execute this file from the command line by typing:
+ *    psql postgres < schema.sql
+*/
+
+DROP DATABASE IF EXISTS airbnb;
```
Comment (JacksonTodd93) [inline schema.sql:5]: I'd pick a more specific database name. We ran into proxy troubles on FEC when everybody's db was named 'zagat', and I can see the same thing happening here.

File: generator.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+const fs = require('fs')
+const faker = require('faker')
+const argv = require('yargs').argv
+const helpers = require('./helpers/helpers.js')
+
+const residences = argv.residences || 10000000
+const filename = argv.output || 'residences.csv'
```
Comment (JacksonTodd93) [inline generator.js:7]: So lines and 6 & 7 define our default values for these parameters in the event that you don't specify a value when invoking your script from the command line?

File: generator.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    do {
+      i--
+      let residence = createResidence();
+      if (i === 0) {
+        writeStream.write(residence, encoding, done)
+      } else {
+        writeStream.write(residence, encoding)
+      }
+    } while (i > 0 && canWrite)
+      if (i > 0 && !canWrite) {
```
Comment (JacksonTodd93) [inline generator.js:30]: This is the one part in here I think I can't follow. Is 'canWrite' a property of your buffer stream that you set to true by default but will get flipped back to false if the buffer is, in fact, full and we can't write?

**Review by JacksonTodd93 — COMMENTED** [review]
Fancy stuff! Everything looks good in general, I left a few comments on specific lines where I had questions

File: helpers/helpers.js (hunk 2 above)
Comment (cinzia-sss) [reply helpers/helpers.js:5]: +1 on this one, how are you planning on completing the photo path?

File: schema.sql (hunk 4 above)
Comment (cinzia-sss) [reply schema.sql:5]: That's a good point @JacksonTodd93  and I made the same choice (mistake?) as @rogen-code. I don't think we're going to run the proxy locally this time tho. It was only a problem when we tried to have all our four modules on our local machine.

**Pull request merged**